apps/emqx_management/src/emqx_mgmt_api.erl

```diff
@@ -707,20 +709,19 @@ format_query_result(
         end,
     #{
         meta => Meta,
-        data => lists:flatten(
-            lists:foldl(
-                fun({Node, Rows}, Acc) ->
-                    [
-                        lists:map(fun(Row) -> exec_format_fun(FmtFun, Node, Row, Opts) end, Rows)
-                        | Acc
-                    ]
-                end,
-                [],
-                RowsAcc
-            )
-        )
+        data => format_query_data(FmtFun, RowsAcc, Opts)
     }.
 
+format_query_data(FmtFun, RowsAcc, Opts) ->
+    %% NOTE: `RowsAcc` is reversed in the node-order, `lists:foldl/3` is correct here.
```

Comment:
make sense to `lists:append(lists:reverse(...))` ?

Reply:
The `FmtFun` needs to run per each row, but the list is `[{Node, [Row]}]`, so I think the current implementation is needed (and also likely the cheapest one).